PWGJE/FlavourJetTasks/AliAnalysisTaskHFSubstructure.cxx

```diff
@@ -347,11 +344,11 @@ Bool_t AliAnalysisTaskHFSubstructure::FillHistograms()
 	        Matched_Truth_Particle_Mother_Label=Matched_Truth_Particle_Mother->GetMother();
 	      }
 	      else break;
-	      delete Matched_Truth_Particle_Mother;
```

Comment:
Such cases (i.e. else break; branches) are easy to overlook - why not using std::unique_ptr<AliAODRecoDecayHF2Prongs>. Furthermore I would be cautious because in principle the AliHFTrackContainer is taking ownership.